tq/basic_upload.go

```diff
@@ -80,7 +86,9 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 		}
 		return nil
 	}
-	var reader lfsapi.ReadSeekCloser = progress.NewBodyWithCallback(f, t.Size, ccb)
+
+	fcount := tools.NewCountingReadSeekCloser(f, stat.Size())
+	var reader lfsapi.ReadSeekCloser = progress.NewBodyWithCallback(fcount, t.Size, ccb)
```

Comment:
Why not just do the counting in the `*bodyWithCallback` that's returned? Why is a new type needed?